tracing-undertow/src/main/java/com/palantir/tracing/undertow/UndertowTracing.java

```diff
@@ -73,6 +75,11 @@ private static void setExchangeState(HttpServerExchange exchange, DetachedSpan d
         // Populate response before proceeding since later operations might commit the response.
         exchange.getResponseHeaders().put(TRACE_ID, traceId);
         exchange.putAttachment(TracingAttachments.IS_SAMPLED, InternalTracers.isSampled(detachedSpan));
+        Optional<String> requestId = InternalTracers.getRequestId(detachedSpan);
+        if (!requestId.isPresent()) {
+            throw new SafeIllegalStateException("No requestId is set", SafeArg.of("span", detachedSpan));
+        }
+        exchange.putAttachment(TracingAttachments.REQUEST_ID, requestId.get());
```

Comment:
I think we should add a response header as well for easier debugging

Reply:
I'm not sure I'm ready to modify the wire api at this point, the traceid is more actionable in the majority of cases as an api consumer while the requestId gives us the ability to do stronger structured analysis.

Reply:
ok, we can add it later if needed